File: JAVA/Kth Smallest.java
class Solution {
    public static int kthSmallest(int[] arr, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        for (int num : arr) {
            minHeap.add(num);
        }
        for (int i = 1; i < k; i++) {
            minHeap.poll();
        }
        return minHeap.poll();
    }
}

File: C++/Kth Smallest.cpp
class Solution {
  public:
    int kthSmallest(vector<int> &arr, int k) {
        int n=arr.size();
        int maxi=*max_element(arr.begin(),arr.end());
        vector<int>v(maxi+1,-1);
        for(int i=0;i<n;i++){
            v[arr[i]]=i;
        }
        int cnt=0;
        for(int i=0;i<=maxi;i++){
            if(v[i]!=-1){
                cnt++;
                if(cnt==k){
                    return i;
                }
            }
        }
        return -1;
    }
};
